crates/scarb-api/src/artifacts.rs

```diff
@@ -92,3 +92,115 @@ fn compile_artifact_at_path(path: &Utf8Path) -> Result<StarknetContractArtifacts
 
     Ok(StarknetContractArtifacts { sierra, casm })
 }
+
+#[cfg(test)]
+mod tests {
+    use std::path::Path;
+
+    use super::*;
+    use crate::ScarbCommand;
+    use camino::Utf8PathBuf;
+    use deserialized::{StarknetArtifacts, StarknetContract, StarknetContractArtifactPaths};
+
+    #[test]
+    fn test_unique_artifacts() {
+        // Mock StarknetArtifactsRepresentation
+        let mock_base_artifacts = HashMap::from([(
+            "contract1".to_string(),
+            (
+                StarknetContractArtifacts {
+                    sierra: "sierra1".to_string(),
+                    casm: "casm1".to_string(),
+                },
+                Utf8PathBuf::from("path1"),
+            ),
+        )]);
+
+        let mock_representation_1 = StarknetArtifactsRepresentation {
+            base_path: Utf8PathBuf::from("mock/path1"),
+            artifacts: StarknetArtifacts {
+                version: 1,
+                contracts: vec![StarknetContract {
+                    id: "1".to_string(),
+                    package_name: "package1".to_string(),
+                    contract_name: "contract1".to_string(),
+                    artifacts: StarknetContractArtifactPaths {
+                        sierra: Utf8PathBuf::from("mock/path1/contract1.sierra"),
+                    },
+                }],
+            },
+        };
+
+        let mock_representation_2 = StarknetArtifactsRepresentation {
+            base_path: Utf8PathBuf::from("mock/path2"),
+            artifacts: StarknetArtifacts {
+                version: 1,
+                contracts: vec![StarknetContract {
+                    id: "2".to_string(),
+                    package_name: "package2".to_string(),
+                    contract_name: "contract2".to_string(),
+                    artifacts: StarknetContractArtifactPaths {
+                        sierra: Utf8PathBuf::from("mock/path2/contract2.sierra"),
+                    },
+                }],
+            },
+        };
+
+        let representations = vec![mock_representation_1, mock_representation_2];
+
+        let result = unique_artifacts(representations, &mock_base_artifacts);
+
+        assert_eq!(result.len(), 1);
+        assert_eq!(result[0].0, "contract2");
+    }
+
+    #[test]
+    fn test_load_contracts_artifacts() {
+        let temp = crate::tests::setup_package("basic_package");
+
+        ScarbCommand::new_with_stdio()
+            .current_dir(temp.path())
+            .arg("build")
+            .run()
+            .unwrap();
```

Comment:
If we want to test creating `StarknetArtifactFiles` with both base and other files, we need scarb version of at least 2.8.3, and we need to build artifacts with the test target

```suggestion
    #[test]
    #[cfg_attr(not(feature = "scarb_2_8_3"), ignore)]
    fn test_load_contracts_artifacts() {
        let temp = crate::tests::setup_package("basic_package");

        ScarbCommand::new_with_stdio()
            .current_dir(temp.path())
            .arg("build")
            .arg("--test")
            .run()
            .unwrap();
```

Reply:
Then you will get two `.test.starknet_artifcats.json` files. One for unittest and one for integrationtest target.